[jsk_pcl_ros/test_extract_indices.cpp] use std::isnan in test_extract_indices

knorth55 opened the pull request.

k-okada commented:
Why travis can not find this error? Or this is just a warning?

C.f. https://github.com/jsk-ros-pkg/jsk_visualization/pull/687#issuecomment-373558702

knorth55 commented:
Maybe, it depends on compiler. I did this test many times on indigo with no failure, but I found this error on kinetic.
Kinetic uses C++11 as default (as my understanding), so it might be the reason, too.

k-okada commented:
but why travis did not fail? for example
https://travis-ci.org/jsk-ros-pkg/jsk_recognition/jobs/353656894

--
◉ Kei Okada


2018-03-16 13:25 GMT+09:00 Shingo Kitagawa <notifications@github.com>:

> Maybe, it depends on compiler. I did this test many times on indigo with
> no failure, but I found this error on kinetic.
> Kinetic uses C++11 as default (as my understanding), so it might be the
> reason, too.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/2251#issuecomment-373599793>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3C_n-u2Tpq7k1QxhO_bsLkGAB-_kks5tez60gaJpZM4Sr4F9>
> .
>

knorth55 commented:
@k-okada Can you merge this PR?

k-okada commented:
What compiler are you using? by default, xenial uses 
```
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.9) 
```
please double check your environment.
```
$ head logs/jsk_pcl_ros/build.cmake.log 
Not searching for unused variables given on the command line.
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
```

knorth55 commented:
my gcc version is same
```
$ gcc --version
gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

k-okada commented:
Then please check your build log very carefully

2018年4月3日(火) 9:42 Shingo Kitagawa <notifications@github.com>:

> my gcc version is same
>
> $ gcc --version
> gcc (Ubuntu 5.4.0-6ubuntu1~16.04.9) 5.4.0 20160609
> Copyright (C) 2015 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/2251#issuecomment-378091750>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3GhfrfhDT4vgQxTcPe2xdSd-m5MAks5tksUogaJpZM4Sr4F9>
> .
>
-- 
--
◉ Kei Okada

knorth55 commented:
It seems that it uses default `cc` and `c++`
```log
Not searching for unused variables given on the command line.
Re-run cmake no build system arguments
[36m--[0m The C compiler identification is GNU 5.4.0
[36m--[0m The CXX compiler identification is GNU 5.4.0
[36m--[0m Check for working C compiler: /usr/bin/cc
[36m--[0m Check for working C compiler: /usr/bin/cc -- works
[36m--[0m Detecting C compiler ABI info
[36m--[0m Detecting C compiler ABI info - done
[36m--[0m Detecting C compile features
[36m--[0m Detecting C compile features - done
[36m--[0m Check for working CXX compiler: /usr/bin/c++
[36m--[0m Check for working CXX compiler: /usr/bin/c++ -- works
[36m--[0m Detecting CXX compiler ABI info
[36m--[0m Detecting CXX compiler ABI info - done
[36m--[0m Detecting CXX compile features
[36m--[0m Detecting CXX compile features - done
[36m--[0m Performing Test COMPILER_SUPPORTS_CXX11
[36m--[0m Performing Test COMPILER_SUPPORTS_CXX11 - Success
[36m--[0m Performing Test COMPILER_SUPPORTS_CXX0X
[36m--[0m Performing Test COMPILER_SUPPORTS_CXX0X - Success
[36m--[0m Using CATKIN_DEVEL_PREFIX: /home/shingo/ros/kinetic/devel/.private/jsk_pcl_ros
[36m--[0m Using CMAKE_PREFIX_PATH: /home/shingo/ros/kinetic/devel;/opt/ros/kinetic
[36m--[0m This workspace overlays: /home/shingo/ros/kinetic/devel;/opt/ros/kinetic
[36m--[0m Found PythonInterp: /usr/bin/python (found version "2.7.12")
[36m--[0m Using PYTHON_EXECUTABLE: /usr/bin/python
[36m--[0m Using Debian Python package layout
[36m--[0m Using empy: /usr/bin/empy
[36m--[0m Using CATKIN_ENABLE_TESTING: ON
[36m--[0m Call enable_testing()
[36m--[0m Using CATKIN_TEST_RESULTS_DIR: /home/shingo/ros/kinetic/build/jsk_pcl_ros/test_results
[36m--[0m Found gmock sources under '/usr/src/gmock': gmock will be built
[36m--[0m Looking for pthread.h
[36m--[0m Looking for pthread.h - found
[36m--[0m Looking for pthread_create
[36m--[0m Looking for pthread_create - not found
[36m--[0m Looking for pthread_create in pthreads
[36m--[0m Looking for pthread_create in pthreads - not found
[36m--[0m Looking for pthread_create in pthread
[36m--[0m Looking for pthread_create in pthread - found
[36m--[0m Found Threads: TRUE
[36m--[0m Found gtest sources under '/usr/src/gmock': gtests will be built
[36m--[0m Using Python nosetests: /usr/local/bin/nosetests-2.7
[36m--[0m catkin 0.7.11
[36m--[0m Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
[36m--[0m Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
[36m--[0m Checking for module 'eigen3'
[36m--[0m   Found eigen3, version 3.2.92
[36m--[0m Found eigen: /usr/include/eigen3
[36m--[0m Boost version: 1.58.0
[36m--[0m Found the following Boost libraries:
[36m--[0m   system
[36m--[0m   filesystem
[36m--[0m   thread
[36m--[0m   date_time
[36m--[0m   iostreams
[36m--[0m   serialization
[36m--[0m   chrono
[36m--[0m   atomic
[36m--[0m   regex
[36m--[0m Checking for module 'libopenni'
[36m--[0m   Found libopenni, version 1.5.4.0
[36m--[0m Found openni: /usr/lib/libOpenNI.so
[36m--[0m Checking for module 'libopenni2'
[36m--[0m   Found libopenni2, version 2.2.0.3
[36m--[0m Found OpenNI2: /usr/lib/libOpenNI2.so
[36m--[0m Could NOT find ensenso (missing:  ENSENSO_LIBRARY)
** WARNING ** io features related to ensenso will be disabled
[36m--[0m Could NOT find DAVIDSDK (missing:  DAVIDSDK_LIBRARY DAVIDSDK_INCLUDE_DIR)
** WARNING ** io features related to davidSDK will be disabled
[36m--[0m Could NOT find DSSDK (missing:  _DSSDK_LIBRARIES)
** WARNING ** io features related to dssdk will be disabled
** WARNING ** io features related to pcap will be disabled
** WARNING ** io features related to png will be disabled
[36m--[0m The imported target "vtkRenderingPythonTkWidgets" references the file
   "/usr/lib/x86_64-linux-gnu/libvtkRenderingPythonTkWidgets.so"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
but not all the files it references.

[36m--[0m The imported target "vtk" references the file
   "/usr/bin/vtk"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
but not all the files it references.

[36m--[0m Found libusb-1.0: /usr/include
** WARNING ** io features related to libusb-1.0 will be disabled
[36m--[0m Checking for module 'flann'
[36m--[0m   Found flann, version 1.8.4
[36m--[0m Found Flann: /usr/lib/x86_64-linux-gnu/libflann_cpp_s.a
[36m--[0m Could NOT find ensenso (missing:  ENSENSO_LIBRARY)
** WARNING ** visualization features related to ensenso will be disabled
[36m--[0m Could NOT find DAVIDSDK (missing:  DAVIDSDK_LIBRARY DAVIDSDK_INCLUDE_DIR)
** WARNING ** visualization features related to davidSDK will be disabled
[36m--[0m Could NOT find DSSDK (missing:  _DSSDK_LIBRARIES)
** WARNING ** visualization features related to dssdk will be disabled
[36m--[0m Could NOT find RSSDK (missing:  _RSSDK_LIBRARIES)
** WARNING ** visualization features related to rssdk will be disabled
[36m--[0m Found qhull: /usr/lib/x86_64-linux-gnu/libqhull.so
[36m--[0m looking for PCL_COMMON
[36m--[0m Found PCL_COMMON: /usr/local/lib/libpcl_common.so
[36m--[0m looking for PCL_OCTREE
[36m--[0m Found PCL_OCTREE: /usr/local/lib/libpcl_octree.so
[36m--[0m looking for PCL_IO
[36m--[0m Found PCL_IO: /usr/local/lib/libpcl_io.so
[36m--[0m looking for PCL_KDTREE
[36m--[0m Found PCL_KDTREE: /usr/local/lib/libpcl_kdtree.so
[36m--[0m looking for PCL_SEARCH
[36m--[0m Found PCL_SEARCH: /usr/local/lib/libpcl_search.so
[36m--[0m looking for PCL_SAMPLE_CONSENSUS
[36m--[0m Found PCL_SAMPLE_CONSENSUS: /usr/local/lib/libpcl_sample_consensus.so
[36m--[0m looking for PCL_FILTERS
[36m--[0m Found PCL_FILTERS: /usr/local/lib/libpcl_filters.so
[36m--[0m looking for PCL_2D
[36m--[0m Found PCL_2D: /usr/local/include/pcl-1.8
[36m--[0m looking for PCL_FEATURES
[36m--[0m Found PCL_FEATURES: /usr/local/lib/libpcl_features.so
[36m--[0m looking for PCL_KEYPOINTS
[36m--[0m Found PCL_KEYPOINTS: /usr/local/lib/libpcl_keypoints.so
[36m--[0m looking for PCL_GEOMETRY
[36m--[0m Found PCL_GEOMETRY: /usr/local/include/pcl-1.8
[36m--[0m looking for PCL_ML
[36m--[0m Found PCL_ML: /usr/local/lib/libpcl_ml.so
[36m--[0m looking for PCL_SEGMENTATION
[36m--[0m Found PCL_SEGMENTATION: /usr/local/lib/libpcl_segmentation.so
[36m--[0m looking for PCL_VISUALIZATION
[36m--[0m Found PCL_VISUALIZATION: /usr/local/lib/libpcl_visualization.so
[36m--[0m looking for PCL_OUTOFCORE
[36m--[0m Found PCL_OUTOFCORE: /usr/local/lib/libpcl_outofcore.so
[36m--[0m looking for PCL_STEREO
[36m--[0m Found PCL_STEREO: /usr/local/lib/libpcl_stereo.so
[36m--[0m looking for PCL_SURFACE
[36m--[0m Found PCL_SURFACE: /usr/local/lib/libpcl_surface.so
[36m--[0m looking for PCL_REGISTRATION
[36m--[0m Found PCL_REGISTRATION: /usr/local/lib/libpcl_registration.so
[36m--[0m looking for PCL_TRACKING
[36m--[0m Found PCL_TRACKING: /usr/local/lib/libpcl_tracking.so
[36m--[0m looking for PCL_RECOGNITION
[36m--[0m Found PCL_RECOGNITION: /usr/local/lib/libpcl_recognition.so
[36m--[0m looking for PCL_PEOPLE
[36m--[0m Found PCL_PEOPLE: /usr/local/lib/libpcl_people.so
[36m--[0m Found PCL: /usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_iostreams.so;/usr/lib/x86_64-linux-gnu/libboost_serialization.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libboost_regex.so;/usr/lib/x86_64-linux-gnu/libpthread.so;optimized;/usr/local/lib/libpcl_common.so;debug;/usr/local/lib/libpcl_common.so;optimized;/usr/local/lib/libpcl_octree.so;debug;/usr/local/lib/libpcl_octree.so;/usr/lib/libOpenNI.so;/usr/lib/libOpenNI2.so;vtkViewsCore;vtkInteractionWidgets;vtkFiltersHybrid;vtkFiltersGeneral;vtkCommonComputationalGeometry;vtkCommonDataModel;vtkCommonMath;vtkCommonCore;vtksys;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkFiltersCore;vtkCommonExecutionModel;vtkImagingSources;vtkImagingCore;vtkRenderingCore;vtkCommonColor;vtkFiltersExtraction;vtkFiltersStatistics;vtkImagingFourier;vtkalglib;vtkFiltersGeometry;vtkFiltersSources;vtkFiltersModeling;vtkImagingGeneral;vtkImagingHybrid;vtkIOImage;vtkDICOMParser;vtkIOCore;/usr/lib/x86_64-linux-gnu/libz.so;vtkmetaio;/usr/lib/x86_64-linux-gnu/libjpeg.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libtiff.so;vtkInteractionStyle;vtkRenderingAnnotation;vtkImagingColor;vtkRenderingFreeType;/usr/lib/x86_64-linux-gnu/libfreetype.so;vtkftgl;vtkRenderingVolume;vtkInteractionImage;vtkRenderingOpenGL;vtkRenderingParallel;vtkFiltersParallel;vtkParallelCore;vtkIOLegacy;vtkIOExodus;vtkIOXML;vtkIOGeometry;/usr/lib/x86_64-linux-gnu/libjsoncpp.so;vtkIOXMLParser;/usr/lib/x86_64-linux-gnu/libexpat.so;vtkexoIIc;/usr/lib/x86_64-linux-gnu/libnetcdf_c++.so;/usr/lib/x86_64-linux-gnu/libnetcdf.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5_hl.so;/usr/lib/openmpi/lib/libmpi.so;vtkRenderingFreeTypeFontConfig;vtkIOParallelLSDyna;vtkIOLSDyna;vtkTestingRendering;vtkInfovisCore;vtkIOParallelNetCDF;vtkParallelMPI;vtkIOVPIC;VPIC;vtkIOMPIParallel;vtkFiltersAMR;vtkWrappingTools;vtkRenderingLOD;vtkRenderingFreeTypeOpenGL;vtkRenderingLIC;/usr/lib/x86_64-linux-gnu/libtheoraenc.so;/usr/lib/x86_64-linux-gnu/libtheoradec.so;/usr/lib/x86_64-linux-gnu/libogg.so;vtkInfovisLayout;vtkIOPLY;vtkFiltersProgrammable;/usr/lib/libgl2ps.so;vtkFiltersReebGraph;vtkIOFFMPEG;vtkIOMovie;vtkIOEnSight;vtkproj4;vtkIONetCDF;vtkRenderingVolumeOpenGL;vtkFiltersParallelImaging;vtkFiltersImaging;vtkIOInfovis;/usr/lib/x86_64-linux-gnu/libxml2.so;vtkImagingMath;vtkFiltersSelection;vtkIOExport;vtkRenderingContext2D;vtkRenderingGL2PS;vtkRenderingContextOpenGL;vtkRenderingLabel;vtkIOMySQL;vtkIOSQL;sqlite3;vtkxdmf2;vtkRenderingParallelLIC;vtkRenderingQt;vtkFiltersTexture;vtkGUISupportQt;vtkIOImport;vtkWrappingPythonCore;/usr/lib/x86_64-linux-gnu/libpython2.7.so;vtkIOODBC;vtkIOPostgreSQL;vtkRenderingImage;verdict;vtkFiltersParallelStatistics;vtkImagingMorphological;vtkRenderingMatplotlib;vtkPythonInterpreter;vtkFiltersHyperTree;vtkChartsCore;vtkViewsQt;vtkViewsInfovis;vtkGUISupportQtWebkit;vtkTestingIOSQL;vtkFiltersParallelMPI;vtkWrappingJava;vtkIOGDAL;vtkGeovisCore;vtkInfovisBoostGraphAlgorithms;vtkFiltersParallelFlowPaths;vtkFiltersFlowPaths;vtkImagingStatistics;vtkIOAMR;vtkFiltersGeneric;vtkGUISupportQtSQL;vtkImagingStencil;vtkIOMINC;vtkLocalExample;vtkFiltersParallelGeometry;vtkIOParallelExodus;vtkViewsGeovis;vtkIOMPIImage;vtkTestingGenericBridge;vtkIOVideo;vtkParallelMPI4Py;vtkGUISupportQtOpenGL;vtkFiltersVerdict;vtkIOParallelXML;vtkDomainsChemistry;vtkIOXdmf2;vtkIOParallel;vtkRenderingExternal;vtkFiltersPython;vtkIOGeoJSON;vtkFiltersSMP;vtkViewsContext2D;optimized;/usr/local/lib/libpcl_io.so;debug;/usr/local/lib/libpcl_io.so;optimized;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;debug;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;optimized;/usr/local/lib/libpcl_kdtree.so;debug;/usr/local/lib/libpcl_kdtree.so;optimized;/usr/local/lib/libpcl_search.so;debug;/usr/local/lib/libpcl_search.so;optimized;/usr/local/lib/libpcl_sample_consensus.so;debug;/usr/local/lib/libpcl_sample_consensus.so;optimized;/usr/local/lib/libpcl_filters.so;debug;/usr/local/lib/libpcl_filters.so;optimized;/usr/local/lib/libpcl_features.so;debug;/usr/local/lib/libpcl_features.so;optimized;/usr/local/lib/libpcl_keypoints.so;debug;/usr/local/lib/libpcl_keypoints.so;optimized;/usr/local/lib/libpcl_ml.so;debug;/usr/local/lib/libpcl_ml.so;optimized;/usr/local/lib/libpcl_segmentation.so;debug;/usr/local/lib/libpcl_segmentation.so;optimized;/usr/local/lib/libpcl_visualization.so;debug;/usr/local/lib/libpcl_visualization.so;optimized;/usr/local/lib/libpcl_outofcore.so;debug;/usr/local/lib/libpcl_outofcore.so;optimized;/usr/local/lib/libpcl_stereo.so;debug;/usr/local/lib/libpcl_stereo.so;optimized;/usr/lib/x86_64-linux-gnu/libqhull.so;debug;/usr/lib/x86_64-linux-gnu/libqhull.so;optimized;/usr/local/lib/libpcl_surface.so;debug;/usr/local/lib/libpcl_surface.so;optimized;/usr/local/lib/libpcl_registration.so;debug;/usr/local/lib/libpcl_registration.so;optimized;/usr/local/lib/libpcl_tracking.so;debug;/usr/local/lib/libpcl_tracking.so;optimized;/usr/local/lib/libpcl_recognition.so;debug;/usr/local/lib/libpcl_recognition.so;optimized;/usr/local/lib/libpcl_people.so;debug;/usr/local/lib/libpcl_people.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_iostreams.so;/usr/lib/x86_64-linux-gnu/libboost_serialization.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libboost_regex.so;/usr/lib/x86_64-linux-gnu/libpthread.so;optimized;/usr/lib/x86_64-linux-gnu/libqhull.so;debug;/usr/lib/x86_64-linux-gnu/libqhull.so;/usr/lib/libOpenNI.so;/usr/lib/libOpenNI2.so;optimized;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;debug;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;vtkViewsCore;vtkInteractionWidgets;vtkFiltersHybrid;vtkFiltersGeneral;vtkCommonComputationalGeometry;vtkCommonDataModel;vtkCommonMath;vtkCommonCore;vtksys;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkFiltersCore;vtkCommonExecutionModel;vtkImagingSources;vtkImagingCore;vtkRenderingCore;vtkCommonColor;vtkFiltersExtraction;vtkFiltersStatistics;vtkImagingFourier;vtkalglib;vtkFiltersGeometry;vtkFiltersSources;vtkFiltersModeling;vtkImagingGeneral;vtkImagingHybrid;vtkIOImage;vtkDICOMParser;vtkIOCore;/usr/lib/x86_64-linux-gnu/libz.so;vtkmetaio;/usr/lib/x86_64-linux-gnu/libjpeg.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libtiff.so;vtkInteractionStyle;vtkRenderingAnnotation;vtkImagingColor;vtkRenderingFreeType;/usr/lib/x86_64-linux-gnu/libfreetype.so;vtkftgl;vtkRenderingVolume;vtkInteractionImage;vtkRenderingOpenGL;vtkRenderingParallel;vtkFiltersParallel;vtkParallelCore;vtkIOLegacy;vtkIOExodus;vtkIOXML;vtkIOGeometry;/usr/lib/x86_64-linux-gnu/libjsoncpp.so;vtkIOXMLParser;/usr/lib/x86_64-linux-gnu/libexpat.so;vtkexoIIc;/usr/lib/x86_64-linux-gnu/libnetcdf_c++.so;/usr/lib/x86_64-linux-gnu/libnetcdf.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5_hl.so;/usr/lib/openmpi/lib/libmpi.so;vtkRenderingFreeTypeFontConfig;vtkIOParallelLSDyna;vtkIOLSDyna;vtkTestingRendering;vtkInfovisCore;vtkIOParallelNetCDF;vtkParallelMPI;vtkIOVPIC;VPIC;vtkIOMPIParallel;vtkFiltersAMR;vtkWrappingTools;vtkRenderingLOD;vtkRenderingFreeTypeOpenGL;vtkRenderingLIC;/usr/lib/x86_64-linux-gnu/libtheoraenc.so;/usr/lib/x86_64-linux-gnu/libtheoradec.so;/usr/lib/x86_64-linux-gnu/libogg.so;vtkInfovisLayout;vtkIOPLY;vtkFiltersProgrammable;/usr/lib/libgl2ps.so;vtkFiltersReebGraph;vtkIOFFMPEG;vtkIOMovie;vtkIOEnSight;vtkproj4;vtkIONetCDF;vtkRenderingVolumeOpenGL;vtkFiltersParallelImaging;vtkFiltersImaging;vtkIOInfovis;/usr/lib/x86_64-linux-gnu/libxml2.so;vtkImagingMath;vtkFiltersSelection;vtkIOExport;vtkRenderingContext2D;vtkRenderingGL2PS;vtkRenderingContextOpenGL;vtkRenderingLabel;vtkIOMySQL;vtkIOSQL;sqlite3;vtkxdmf2;vtkRenderingParallelLIC;vtkRenderingQt;vtkFiltersTexture;vtkGUISupportQt;vtkIOImport;vtkWrappingPythonCore;/usr/lib/x86_64-linux-gnu/libpython2.7.so;vtkIOODBC;vtkIOPostgreSQL;vtkRenderingImage;verdict;vtkFiltersParallelStatistics;vtkImagingMorphological;vtkRenderingMatplotlib;vtkPythonInterpreter;vtkFiltersHyperTree;vtkChartsCore;vtkViewsQt;vtkViewsInfovis;vtkGUISupportQtWebkit;vtkTestingIOSQL;vtkFiltersParallelMPI;vtkWrappingJava;vtkIOGDAL;vtkGeovisCore;vtkInfovisBoostGraphAlgorithms;vtkFiltersParallelFlowPaths;vtkFiltersFlowPaths;vtkImagingStatistics;vtkIOAMR;vtkFiltersGeneric;vtkGUISupportQtSQL;vtkImagingStencil;vtkIOMINC;vtkLocalExample;vtkFiltersParallelGeometry;vtkIOParallelExodus;vtkViewsGeovis;vtkIOMPIImage;vtkTestingGenericBridge;vtkIOVideo;vtkParallelMPI4Py;vtkGUISupportQtOpenGL;vtkFiltersVerdict;vtkIOParallelXML;vtkDomainsChemistry;vtkIOXdmf2;vtkIOParallel;vtkRenderingExternal;vtkFiltersPython;vtkIOGeoJSON;vtkFiltersSMP;vtkViewsContext2D
[36m--[0m Try OpenMP C flag = [-fopenmp]
[36m--[0m Performing Test OpenMP_FLAG_DETECTED
[36m--[0m Performing Test OpenMP_FLAG_DETECTED - Success
[36m--[0m Try OpenMP CXX flag = [-fopenmp]
[36m--[0m Performing Test OpenMP_FLAG_DETECTED
[36m--[0m Performing Test OpenMP_FLAG_DETECTED - Success
[36m--[0m Found OpenMP: -fopenmp
[36m--[0m Could NOT find PCL_GPU_KINFU_LARGE_SCALE (missing:  PCL_GPU_KINFU_LARGE_SCALE_LIBRARY PCL_GPU_KINFU_LARGE_SCALE_INCLUDE_DIR)
[36m--[0m Checking for module 'yaml-cpp'
[36m--[0m   Found yaml-cpp, version 0.5.2
[36m--[0m Found OpenCV: /opt/ros/kinetic (found version "3.3.1") found components:  core imgproc
DEPRECATION WARNING: nodelet 'jsk_pcl/NormalFlipToFrame' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/CentroidPublisher' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformBoundingBox' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformBoundingBoxArray' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/NormalConcatenater' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayDistanceLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayAreaLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayAngleLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayFootAngleLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PoseWithCovarianceStampedToGaussianPointCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/SphericalPointCloudSimulator' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonFlipper' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonPointsSampler' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonMagnifier' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PlanarPointCloudSimulator' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneRejector' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PointCloudToClusterPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/StaticPolygonArrayPublisher' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayTransformer' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PointCloudToSTL' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonAppender' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/DelayPointCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/DepthImageError' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayWrapper' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayUnwrapper' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/ColorizeDistanceFromPlane' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/ColorizeHeight2DMapping' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneReasoner' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/TransformPointcloudInBoundingBox' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PointIndicesToMaskImage' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/MaskImageToDepthConsideredMaskImage' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/MaskImageToPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/OrganizedPointCloudToPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneConcatenator' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/AddPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
DEPRECATION WARNING: nodelet 'jsk_pcl/PCDReaderWithPose' is moved to upstream package 'jsk_pcl_ros_utils'.
CMake Warning (dev) at CMakeLists.txt:390 (message):
  robot_self_filter is NOT found.  some nodelet depending on
  robot_self_filter will not be compiled.
This warning is for project developers.  Use -Wno-dev to suppress it.

flags: -std=c++11  -fopenmp
[33m[1mCMake Warning[0m at /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:166 (message):
  catkin_package() DEPENDS on 'pcl' but neither 'pcl_INCLUDE_DIRS' nor
  'pcl_LIBRARIES' is defined.
[36mCall Stack (most recent call first):[0m
  /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:102 (_catkin_package)
  CMakeLists.txt:441 (catkin_package)


[36m--[0m Configuring done
CMake Warning (dev) at CMakeLists.txt:413 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros" does not
  exist.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at CMakeLists.txt:414 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros_base"
  does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at CMakeLists.txt:415 (add_dependencies):
  Policy CMP0046 is not set: Error on non-existent dependency in
  add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
  Use the cmake_policy command to set the policy and suppress this warning.

  The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros_moveit"
  does not exist.
This warning is for project developers.  Use -Wno-dev to suppress it.

[36m--[0m Generating done
[36m--[0m Build files have been written to: /home/shingo/ros/kinetic/build/jsk_pcl_ros
```

k-okada commented:
Run with -vi, check command line that fails, re-run that command manually,
sometimes gcc -E helps

2018年4月3日(火) 9:51 Shingo Kitagawa <notifications@github.com>:

> It seems that it uses default cc and c++
>
> Not searching for unused variables given on the command line.
> Re-run cmake no build system arguments
> �[36m--�[0m The C compiler identification is GNU 5.4.0
> �[36m--�[0m The CXX compiler identification is GNU 5.4.0
> �[36m--�[0m Check for working C compiler: /usr/bin/cc
> �[36m--�[0m Check for working C compiler: /usr/bin/cc -- works
> �[36m--�[0m Detecting C compiler ABI info
> �[36m--�[0m Detecting C compiler ABI info - done
> �[36m--�[0m Detecting C compile features
> �[36m--�[0m Detecting C compile features - done
> �[36m--�[0m Check for working CXX compiler: /usr/bin/c++
> �[36m--�[0m Check for working CXX compiler: /usr/bin/c++ -- works
> �[36m--�[0m Detecting CXX compiler ABI info
> �[36m--�[0m Detecting CXX compiler ABI info - done
> �[36m--�[0m Detecting CXX compile features
> �[36m--�[0m Detecting CXX compile features - done
> �[36m--�[0m Performing Test COMPILER_SUPPORTS_CXX11
> �[36m--�[0m Performing Test COMPILER_SUPPORTS_CXX11 - Success
> �[36m--�[0m Performing Test COMPILER_SUPPORTS_CXX0X
> �[36m--�[0m Performing Test COMPILER_SUPPORTS_CXX0X - Success
> �[36m--�[0m Using CATKIN_DEVEL_PREFIX: /home/shingo/ros/kinetic/devel/.private/jsk_pcl_ros
> �[36m--�[0m Using CMAKE_PREFIX_PATH: /home/shingo/ros/kinetic/devel;/opt/ros/kinetic
> �[36m--�[0m This workspace overlays: /home/shingo/ros/kinetic/devel;/opt/ros/kinetic
> �[36m--�[0m Found PythonInterp: /usr/bin/python (found version "2.7.12")
> �[36m--�[0m Using PYTHON_EXECUTABLE: /usr/bin/python
> �[36m--�[0m Using Debian Python package layout
> �[36m--�[0m Using empy: /usr/bin/empy
> �[36m--�[0m Using CATKIN_ENABLE_TESTING: ON
> �[36m--�[0m Call enable_testing()
> �[36m--�[0m Using CATKIN_TEST_RESULTS_DIR: /home/shingo/ros/kinetic/build/jsk_pcl_ros/test_results
> �[36m--�[0m Found gmock sources under '/usr/src/gmock': gmock will be built
> �[36m--�[0m Looking for pthread.h
> �[36m--�[0m Looking for pthread.h - found
> �[36m--�[0m Looking for pthread_create
> �[36m--�[0m Looking for pthread_create - not found
> �[36m--�[0m Looking for pthread_create in pthreads
> �[36m--�[0m Looking for pthread_create in pthreads - not found
> �[36m--�[0m Looking for pthread_create in pthread
> �[36m--�[0m Looking for pthread_create in pthread - found
> �[36m--�[0m Found Threads: TRUE
> �[36m--�[0m Found gtest sources under '/usr/src/gmock': gtests will be built
> �[36m--�[0m Using Python nosetests: /usr/local/bin/nosetests-2.7
> �[36m--�[0m catkin 0.7.11
> �[36m--�[0m Using these message generators: gencpp;geneus;genlisp;gennodejs;genpy
> �[36m--�[0m Found PkgConfig: /usr/bin/pkg-config (found version "0.29.1")
> �[36m--�[0m Checking for module 'eigen3'
> �[36m--�[0m   Found eigen3, version 3.2.92
> �[36m--�[0m Found eigen: /usr/include/eigen3
> �[36m--�[0m Boost version: 1.58.0
> �[36m--�[0m Found the following Boost libraries:
> �[36m--�[0m   system
> �[36m--�[0m   filesystem
> �[36m--�[0m   thread
> �[36m--�[0m   date_time
> �[36m--�[0m   iostreams
> �[36m--�[0m   serialization
> �[36m--�[0m   chrono
> �[36m--�[0m   atomic
> �[36m--�[0m   regex
> �[36m--�[0m Checking for module 'libopenni'
> �[36m--�[0m   Found libopenni, version 1.5.4.0
> �[36m--�[0m Found openni: /usr/lib/libOpenNI.so
> �[36m--�[0m Checking for module 'libopenni2'
> �[36m--�[0m   Found libopenni2, version 2.2.0.3
> �[36m--�[0m Found OpenNI2: /usr/lib/libOpenNI2.so
> �[36m--�[0m Could NOT find ensenso (missing:  ENSENSO_LIBRARY)
> ** WARNING ** io features related to ensenso will be disabled
> �[36m--�[0m Could NOT find DAVIDSDK (missing:  DAVIDSDK_LIBRARY DAVIDSDK_INCLUDE_DIR)
> ** WARNING ** io features related to davidSDK will be disabled
> �[36m--�[0m Could NOT find DSSDK (missing:  _DSSDK_LIBRARIES)
> ** WARNING ** io features related to dssdk will be disabled
> ** WARNING ** io features related to pcap will be disabled
> ** WARNING ** io features related to png will be disabled
> �[36m--�[0m The imported target "vtkRenderingPythonTkWidgets" references the file
>    "/usr/lib/x86_64-linux-gnu/libvtkRenderingPythonTkWidgets.so"
> but this file does not exist.  Possible reasons include:
> * The file was deleted, renamed, or moved to another location.
> * An install or uninstall procedure did not complete successfully.
> * The installation package was faulty and contained
>    "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
> but not all the files it references.
>
> �[36m--�[0m The imported target "vtk" references the file
>    "/usr/bin/vtk"
> but this file does not exist.  Possible reasons include:
> * The file was deleted, renamed, or moved to another location.
> * An install or uninstall procedure did not complete successfully.
> * The installation package was faulty and contained
>    "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
> but not all the files it references.
>
> �[36m--�[0m Found libusb-1.0: /usr/include
> ** WARNING ** io features related to libusb-1.0 will be disabled
> �[36m--�[0m Checking for module 'flann'
> �[36m--�[0m   Found flann, version 1.8.4
> �[36m--�[0m Found Flann: /usr/lib/x86_64-linux-gnu/libflann_cpp_s.a
> �[36m--�[0m Could NOT find ensenso (missing:  ENSENSO_LIBRARY)
> ** WARNING ** visualization features related to ensenso will be disabled
> �[36m--�[0m Could NOT find DAVIDSDK (missing:  DAVIDSDK_LIBRARY DAVIDSDK_INCLUDE_DIR)
> ** WARNING ** visualization features related to davidSDK will be disabled
> �[36m--�[0m Could NOT find DSSDK (missing:  _DSSDK_LIBRARIES)
> ** WARNING ** visualization features related to dssdk will be disabled
> �[36m--�[0m Could NOT find RSSDK (missing:  _RSSDK_LIBRARIES)
> ** WARNING ** visualization features related to rssdk will be disabled
> �[36m--�[0m Found qhull: /usr/lib/x86_64-linux-gnu/libqhull.so
> �[36m--�[0m looking for PCL_COMMON
> �[36m--�[0m Found PCL_COMMON: /usr/local/lib/libpcl_common.so
> �[36m--�[0m looking for PCL_OCTREE
> �[36m--�[0m Found PCL_OCTREE: /usr/local/lib/libpcl_octree.so
> �[36m--�[0m looking for PCL_IO
> �[36m--�[0m Found PCL_IO: /usr/local/lib/libpcl_io.so
> �[36m--�[0m looking for PCL_KDTREE
> �[36m--�[0m Found PCL_KDTREE: /usr/local/lib/libpcl_kdtree.so
> �[36m--�[0m looking for PCL_SEARCH
> �[36m--�[0m Found PCL_SEARCH: /usr/local/lib/libpcl_search.so
> �[36m--�[0m looking for PCL_SAMPLE_CONSENSUS
> �[36m--�[0m Found PCL_SAMPLE_CONSENSUS: /usr/local/lib/libpcl_sample_consensus.so
> �[36m--�[0m looking for PCL_FILTERS
> �[36m--�[0m Found PCL_FILTERS: /usr/local/lib/libpcl_filters.so
> �[36m--�[0m looking for PCL_2D
> �[36m--�[0m Found PCL_2D: /usr/local/include/pcl-1.8
> �[36m--�[0m looking for PCL_FEATURES
> �[36m--�[0m Found PCL_FEATURES: /usr/local/lib/libpcl_features.so
> �[36m--�[0m looking for PCL_KEYPOINTS
> �[36m--�[0m Found PCL_KEYPOINTS: /usr/local/lib/libpcl_keypoints.so
> �[36m--�[0m looking for PCL_GEOMETRY
> �[36m--�[0m Found PCL_GEOMETRY: /usr/local/include/pcl-1.8
> �[36m--�[0m looking for PCL_ML
> �[36m--�[0m Found PCL_ML: /usr/local/lib/libpcl_ml.so
> �[36m--�[0m looking for PCL_SEGMENTATION
> �[36m--�[0m Found PCL_SEGMENTATION: /usr/local/lib/libpcl_segmentation.so
> �[36m--�[0m looking for PCL_VISUALIZATION
> �[36m--�[0m Found PCL_VISUALIZATION: /usr/local/lib/libpcl_visualization.so
> �[36m--�[0m looking for PCL_OUTOFCORE
> �[36m--�[0m Found PCL_OUTOFCORE: /usr/local/lib/libpcl_outofcore.so
> �[36m--�[0m looking for PCL_STEREO
> �[36m--�[0m Found PCL_STEREO: /usr/local/lib/libpcl_stereo.so
> �[36m--�[0m looking for PCL_SURFACE
> �[36m--�[0m Found PCL_SURFACE: /usr/local/lib/libpcl_surface.so
> �[36m--�[0m looking for PCL_REGISTRATION
> �[36m--�[0m Found PCL_REGISTRATION: /usr/local/lib/libpcl_registration.so
> �[36m--�[0m looking for PCL_TRACKING
> �[36m--�[0m Found PCL_TRACKING: /usr/local/lib/libpcl_tracking.so
> �[36m--�[0m looking for PCL_RECOGNITION
> �[36m--�[0m Found PCL_RECOGNITION: /usr/local/lib/libpcl_recognition.so
> �[36m--�[0m looking for PCL_PEOPLE
> �[36m--�[0m Found PCL_PEOPLE: /usr/local/lib/libpcl_people.so
> �[36m--�[0m Found PCL: /usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_iostreams.so;/usr/lib/x86_64-linux-gnu/libboost_serialization.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libboost_regex.so;/usr/lib/x86_64-linux-gnu/libpthread.so;optimized;/usr/local/lib/libpcl_common.so;debug;/usr/local/lib/libpcl_common.so;optimized;/usr/local/lib/libpcl_octree.so;debug;/usr/local/lib/libpcl_octree.so;/usr/lib/libOpenNI.so;/usr/lib/libOpenNI2.so;vtkViewsCore;vtkInteractionWidgets;vtkFiltersHybrid;vtkFiltersGeneral;vtkCommonComputationalGeometry;vtkCommonDataModel;vtkCommonMath;vtkCommonCore;vtksys;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkFiltersCore;vtkCommonExecutionModel;vtkImagingSources;vtkImagingCore;vtkRenderingCore;vtkCommonColor;vtkFiltersExtraction;vtkFiltersStatistics;vtkImagingFourier;vtkalglib;vtkFiltersGeometry;vtkFiltersSources;vtkFiltersModeling;vtkImagingGeneral;vtkImagingHybrid;vtkIOImage;vtkDICOMParser;vtkIOCore;/usr/lib/x86_64-linux-gnu/libz.so;vtkmetaio;/usr/lib/x86_64-linux-gnu/libjpeg.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libtiff.so;vtkInteractionStyle;vtkRenderingAnnotation;vtkImagingColor;vtkRenderingFreeType;/usr/lib/x86_64-linux-gnu/libfreetype.so;vtkftgl;vtkRenderingVolume;vtkInteractionImage;vtkRenderingOpenGL;vtkRenderingParallel;vtkFiltersParallel;vtkParallelCore;vtkIOLegacy;vtkIOExodus;vtkIOXML;vtkIOGeometry;/usr/lib/x86_64-linux-gnu/libjsoncpp.so;vtkIOXMLParser;/usr/lib/x86_64-linux-gnu/libexpat.so;vtkexoIIc;/usr/lib/x86_64-linux-gnu/libnetcdf_c++.so;/usr/lib/x86_64-linux-gnu/libnetcdf.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5_hl.so;/usr/lib/openmpi/lib/libmpi.so;vtkRenderingFreeTypeFontConfig;vtkIOParallelLSDyna;vtkIOLSDyna;vtkTestingRendering;vtkInfovisCore;vtkIOParallelNetCDF;vtkParallelMPI;vtkIOVPIC;VPIC;vtkIOMPIParallel;vtkFiltersAMR;vtkWrappingTools;vtkRenderingLOD;vtkRenderingFreeTypeOpenGL;vtkRenderingLIC;/usr/lib/x86_64-linux-gnu/libtheoraenc.so;/usr/lib/x86_64-linux-gnu/libtheoradec.so;/usr/lib/x86_64-linux-gnu/libogg.so;vtkInfovisLayout;vtkIOPLY;vtkFiltersProgrammable;/usr/lib/libgl2ps.so;vtkFiltersReebGraph;vtkIOFFMPEG;vtkIOMovie;vtkIOEnSight;vtkproj4;vtkIONetCDF;vtkRenderingVolumeOpenGL;vtkFiltersParallelImaging;vtkFiltersImaging;vtkIOInfovis;/usr/lib/x86_64-linux-gnu/libxml2.so;vtkImagingMath;vtkFiltersSelection;vtkIOExport;vtkRenderingContext2D;vtkRenderingGL2PS;vtkRenderingContextOpenGL;vtkRenderingLabel;vtkIOMySQL;vtkIOSQL;sqlite3;vtkxdmf2;vtkRenderingParallelLIC;vtkRenderingQt;vtkFiltersTexture;vtkGUISupportQt;vtkIOImport;vtkWrappingPythonCore;/usr/lib/x86_64-linux-gnu/libpython2.7.so;vtkIOODBC;vtkIOPostgreSQL;vtkRenderingImage;verdict;vtkFiltersParallelStatistics;vtkImagingMorphological;vtkRenderingMatplotlib;vtkPythonInterpreter;vtkFiltersHyperTree;vtkChartsCore;vtkViewsQt;vtkViewsInfovis;vtkGUISupportQtWebkit;vtkTestingIOSQL;vtkFiltersParallelMPI;vtkWrappingJava;vtkIOGDAL;vtkGeovisCore;vtkInfovisBoostGraphAlgorithms;vtkFiltersParallelFlowPaths;vtkFiltersFlowPaths;vtkImagingStatistics;vtkIOAMR;vtkFiltersGeneric;vtkGUISupportQtSQL;vtkImagingStencil;vtkIOMINC;vtkLocalExample;vtkFiltersParallelGeometry;vtkIOParallelExodus;vtkViewsGeovis;vtkIOMPIImage;vtkTestingGenericBridge;vtkIOVideo;vtkParallelMPI4Py;vtkGUISupportQtOpenGL;vtkFiltersVerdict;vtkIOParallelXML;vtkDomainsChemistry;vtkIOXdmf2;vtkIOParallel;vtkRenderingExternal;vtkFiltersPython;vtkIOGeoJSON;vtkFiltersSMP;vtkViewsContext2D;optimized;/usr/local/lib/libpcl_io.so;debug;/usr/local/lib/libpcl_io.so;optimized;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;debug;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;optimized;/usr/local/lib/libpcl_kdtree.so;debug;/usr/local/lib/libpcl_kdtree.so;optimized;/usr/local/lib/libpcl_search.so;debug;/usr/local/lib/libpcl_search.so;optimized;/usr/local/lib/libpcl_sample_consensus.so;debug;/usr/local/lib/libpcl_sample_consensus.so;optimized;/usr/local/lib/libpcl_filters.so;debug;/usr/local/lib/libpcl_filters.so;optimized;/usr/local/lib/libpcl_features.so;debug;/usr/local/lib/libpcl_features.so;optimized;/usr/local/lib/libpcl_keypoints.so;debug;/usr/local/lib/libpcl_keypoints.so;optimized;/usr/local/lib/libpcl_ml.so;debug;/usr/local/lib/libpcl_ml.so;optimized;/usr/local/lib/libpcl_segmentation.so;debug;/usr/local/lib/libpcl_segmentation.so;optimized;/usr/local/lib/libpcl_visualization.so;debug;/usr/local/lib/libpcl_visualization.so;optimized;/usr/local/lib/libpcl_outofcore.so;debug;/usr/local/lib/libpcl_outofcore.so;optimized;/usr/local/lib/libpcl_stereo.so;debug;/usr/local/lib/libpcl_stereo.so;optimized;/usr/lib/x86_64-linux-gnu/libqhull.so;debug;/usr/lib/x86_64-linux-gnu/libqhull.so;optimized;/usr/local/lib/libpcl_surface.so;debug;/usr/local/lib/libpcl_surface.so;optimized;/usr/local/lib/libpcl_registration.so;debug;/usr/local/lib/libpcl_registration.so;optimized;/usr/local/lib/libpcl_tracking.so;debug;/usr/local/lib/libpcl_tracking.so;optimized;/usr/local/lib/libpcl_recognition.so;debug;/usr/local/lib/libpcl_recognition.so;optimized;/usr/local/lib/libpcl_people.so;debug;/usr/local/lib/libpcl_people.so;/usr/lib/x86_64-linux-gnu/libboost_system.so;/usr/lib/x86_64-linux-gnu/libboost_filesystem.so;/usr/lib/x86_64-linux-gnu/libboost_thread.so;/usr/lib/x86_64-linux-gnu/libboost_date_time.so;/usr/lib/x86_64-linux-gnu/libboost_iostreams.so;/usr/lib/x86_64-linux-gnu/libboost_serialization.so;/usr/lib/x86_64-linux-gnu/libboost_chrono.so;/usr/lib/x86_64-linux-gnu/libboost_atomic.so;/usr/lib/x86_64-linux-gnu/libboost_regex.so;/usr/lib/x86_64-linux-gnu/libpthread.so;optimized;/usr/lib/x86_64-linux-gnu/libqhull.so;debug;/usr/lib/x86_64-linux-gnu/libqhull.so;/usr/lib/libOpenNI.so;/usr/lib/libOpenNI2.so;optimized;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;debug;/usr/lib/x86_64-linux-gnu/libflann_cpp_s.a;vtkViewsCore;vtkInteractionWidgets;vtkFiltersHybrid;vtkFiltersGeneral;vtkCommonComputationalGeometry;vtkCommonDataModel;vtkCommonMath;vtkCommonCore;vtksys;vtkCommonMisc;vtkCommonSystem;vtkCommonTransforms;vtkFiltersCore;vtkCommonExecutionModel;vtkImagingSources;vtkImagingCore;vtkRenderingCore;vtkCommonColor;vtkFiltersExtraction;vtkFiltersStatistics;vtkImagingFourier;vtkalglib;vtkFiltersGeometry;vtkFiltersSources;vtkFiltersModeling;vtkImagingGeneral;vtkImagingHybrid;vtkIOImage;vtkDICOMParser;vtkIOCore;/usr/lib/x86_64-linux-gnu/libz.so;vtkmetaio;/usr/lib/x86_64-linux-gnu/libjpeg.so;/usr/lib/x86_64-linux-gnu/libpng.so;/usr/lib/x86_64-linux-gnu/libtiff.so;vtkInteractionStyle;vtkRenderingAnnotation;vtkImagingColor;vtkRenderingFreeType;/usr/lib/x86_64-linux-gnu/libfreetype.so;vtkftgl;vtkRenderingVolume;vtkInteractionImage;vtkRenderingOpenGL;vtkRenderingParallel;vtkFiltersParallel;vtkParallelCore;vtkIOLegacy;vtkIOExodus;vtkIOXML;vtkIOGeometry;/usr/lib/x86_64-linux-gnu/libjsoncpp.so;vtkIOXMLParser;/usr/lib/x86_64-linux-gnu/libexpat.so;vtkexoIIc;/usr/lib/x86_64-linux-gnu/libnetcdf_c++.so;/usr/lib/x86_64-linux-gnu/libnetcdf.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5.so;/usr/lib/x86_64-linux-gnu/libsz.so;/usr/lib/x86_64-linux-gnu/libdl.so;/usr/lib/x86_64-linux-gnu/libm.so;/usr/lib/x86_64-linux-gnu/hdf5/openmpi/lib/libhdf5_hl.so;/usr/lib/openmpi/lib/libmpi.so;vtkRenderingFreeTypeFontConfig;vtkIOParallelLSDyna;vtkIOLSDyna;vtkTestingRendering;vtkInfovisCore;vtkIOParallelNetCDF;vtkParallelMPI;vtkIOVPIC;VPIC;vtkIOMPIParallel;vtkFiltersAMR;vtkWrappingTools;vtkRenderingLOD;vtkRenderingFreeTypeOpenGL;vtkRenderingLIC;/usr/lib/x86_64-linux-gnu/libtheoraenc.so;/usr/lib/x86_64-linux-gnu/libtheoradec.so;/usr/lib/x86_64-linux-gnu/libogg.so;vtkInfovisLayout;vtkIOPLY;vtkFiltersProgrammable;/usr/lib/libgl2ps.so;vtkFiltersReebGraph;vtkIOFFMPEG;vtkIOMovie;vtkIOEnSight;vtkproj4;vtkIONetCDF;vtkRenderingVolumeOpenGL;vtkFiltersParallelImaging;vtkFiltersImaging;vtkIOInfovis;/usr/lib/x86_64-linux-gnu/libxml2.so;vtkImagingMath;vtkFiltersSelection;vtkIOExport;vtkRenderingContext2D;vtkRenderingGL2PS;vtkRenderingContextOpenGL;vtkRenderingLabel;vtkIOMySQL;vtkIOSQL;sqlite3;vtkxdmf2;vtkRenderingParallelLIC;vtkRenderingQt;vtkFiltersTexture;vtkGUISupportQt;vtkIOImport;vtkWrappingPythonCore;/usr/lib/x86_64-linux-gnu/libpython2.7.so;vtkIOODBC;vtkIOPostgreSQL;vtkRenderingImage;verdict;vtkFiltersParallelStatistics;vtkImagingMorphological;vtkRenderingMatplotlib;vtkPythonInterpreter;vtkFiltersHyperTree;vtkChartsCore;vtkViewsQt;vtkViewsInfovis;vtkGUISupportQtWebkit;vtkTestingIOSQL;vtkFiltersParallelMPI;vtkWrappingJava;vtkIOGDAL;vtkGeovisCore;vtkInfovisBoostGraphAlgorithms;vtkFiltersParallelFlowPaths;vtkFiltersFlowPaths;vtkImagingStatistics;vtkIOAMR;vtkFiltersGeneric;vtkGUISupportQtSQL;vtkImagingStencil;vtkIOMINC;vtkLocalExample;vtkFiltersParallelGeometry;vtkIOParallelExodus;vtkViewsGeovis;vtkIOMPIImage;vtkTestingGenericBridge;vtkIOVideo;vtkParallelMPI4Py;vtkGUISupportQtOpenGL;vtkFiltersVerdict;vtkIOParallelXML;vtkDomainsChemistry;vtkIOXdmf2;vtkIOParallel;vtkRenderingExternal;vtkFiltersPython;vtkIOGeoJSON;vtkFiltersSMP;vtkViewsContext2D
> �[36m--�[0m Try OpenMP C flag = [-fopenmp]
> �[36m--�[0m Performing Test OpenMP_FLAG_DETECTED
> �[36m--�[0m Performing Test OpenMP_FLAG_DETECTED - Success
> �[36m--�[0m Try OpenMP CXX flag = [-fopenmp]
> �[36m--�[0m Performing Test OpenMP_FLAG_DETECTED
> �[36m--�[0m Performing Test OpenMP_FLAG_DETECTED - Success
> �[36m--�[0m Found OpenMP: -fopenmp
> �[36m--�[0m Could NOT find PCL_GPU_KINFU_LARGE_SCALE (missing:  PCL_GPU_KINFU_LARGE_SCALE_LIBRARY PCL_GPU_KINFU_LARGE_SCALE_INCLUDE_DIR)
> �[36m--�[0m Checking for module 'yaml-cpp'
> �[36m--�[0m   Found yaml-cpp, version 0.5.2
> �[36m--�[0m Found OpenCV: /opt/ros/kinetic (found version "3.3.1") found components:  core imgproc
> DEPRECATION WARNING: nodelet 'jsk_pcl/NormalFlipToFrame' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/CentroidPublisher' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformBoundingBox' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/TfTransformBoundingBoxArray' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/NormalConcatenater' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayDistanceLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayAreaLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayAngleLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayFootAngleLikelihood' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PoseWithCovarianceStampedToGaussianPointCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/SphericalPointCloudSimulator' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonFlipper' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonPointsSampler' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonMagnifier' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PlanarPointCloudSimulator' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneRejector' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PointCloudToClusterPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/StaticPolygonArrayPublisher' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayTransformer' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PointCloudToSTL' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonAppender' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/DelayPointCloud' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/DepthImageError' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayWrapper' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PolygonArrayUnwrapper' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/ColorizeDistanceFromPlane' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/ColorizeHeight2DMapping' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneReasoner' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/TransformPointcloudInBoundingBox' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PointIndicesToMaskImage' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/MaskImageToDepthConsideredMaskImage' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/MaskImageToPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/OrganizedPointCloudToPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PlaneConcatenator' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/AddPointIndices' is moved to upstream package 'jsk_pcl_ros_utils'.
> DEPRECATION WARNING: nodelet 'jsk_pcl/PCDReaderWithPose' is moved to upstream package 'jsk_pcl_ros_utils'.
> CMake Warning (dev) at CMakeLists.txt:390 (message):
>   robot_self_filter is NOT found.  some nodelet depending on
>   robot_self_filter will not be compiled.
> This warning is for project developers.  Use -Wno-dev to suppress it.
>
> flags: -std=c++11  -fopenmp
> �[33m�[1mCMake Warning�[0m at /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:166 (message):
>   catkin_package() DEPENDS on 'pcl' but neither 'pcl_INCLUDE_DIRS' nor
>   'pcl_LIBRARIES' is defined.
> �[36mCall Stack (most recent call first):�[0m
>   /opt/ros/kinetic/share/catkin/cmake/catkin_package.cmake:102 (_catkin_package)
>   CMakeLists.txt:441 (catkin_package)
>
>
> �[36m--�[0m Configuring done
> CMake Warning (dev) at CMakeLists.txt:413 (add_dependencies):
>   Policy CMP0046 is not set: Error on non-existent dependency in
>   add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
>   Use the cmake_policy command to set the policy and suppress this warning.
>
>   The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros" does not
>   exist.
> This warning is for project developers.  Use -Wno-dev to suppress it.
>
> CMake Warning (dev) at CMakeLists.txt:414 (add_dependencies):
>   Policy CMP0046 is not set: Error on non-existent dependency in
>   add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
>   Use the cmake_policy command to set the policy and suppress this warning.
>
>   The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros_base"
>   does not exist.
> This warning is for project developers.  Use -Wno-dev to suppress it.
>
> CMake Warning (dev) at CMakeLists.txt:415 (add_dependencies):
>   Policy CMP0046 is not set: Error on non-existent dependency in
>   add_dependencies.  Run "cmake --help-policy CMP0046" for policy details.
>   Use the cmake_policy command to set the policy and suppress this warning.
>
>   The dependency target "jsk_pcl_ros_gencpp" of target "jsk_pcl_ros_moveit"
>   does not exist.
> This warning is for project developers.  Use -Wno-dev to suppress it.
>
> �[36m--�[0m Generating done
> �[36m--�[0m Build files have been written to: /home/shingo/ros/kinetic/build/jsk_pcl_ros
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/2251#issuecomment-378093205>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3AE-9yf6f5q9NAnP6rYkLAhtaL71ks5tkseagaJpZM4Sr4F9>
> .
>
-- 
--
◉ Kei Okada

wkentaro commented:
> Why travis can not find this error? Or this is just a warning?

1. This raises only on **Kinetic**, in which newer gcc (5.4.0) is used.
2. Test is disabled on **Kinetic**

so travis can not find this.

wkentaro commented:
You can reproduce this error by below:

```
# on Kinetic
roscd jsk_pcl_ros
catkin run_tests --this --no-deps -iv
```

k-okada commented:
thank you for clear explanation!
